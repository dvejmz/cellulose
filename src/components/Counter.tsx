import React from 'react';

export enum QuantityUnitAlignment {
  Left,
  Right,
};

export interface CounterProps {
  id: string;
  name: string;
  quantity: number;
  quantityUnit?: string;
  showDecimals?: boolean;
  quantityUnitAlignment?: QuantityUnitAlignment;
}

const formatQuantity = (quantity: number, showDecimals = false): string => (
  showDecimals
    ? quantity.toFixed(2)
    : quantity.toFixed(0)
);

const Counter: React.FC<CounterProps> = (props: CounterProps) => {
  return (
    <div className="counter margin-top-right" data-test-id={`counter-${props.id}`}>
      <strong className={`counter__name capitalise`}>{props.name}</strong>:{' '}
      {props.quantityUnitAlignment === QuantityUnitAlignment.Left && <span className="counter__unit">{props.quantityUnit}</span>}
      <span className={`counter__value`}>{formatQuantity(props.quantity, props.showDecimals)}</span>{' '} 
      {(props.quantityUnitAlignment === QuantityUnitAlignment.Right || props.quantityUnitAlignment === undefined) && <span className="counter__unit">{props.quantityUnit}</span>}
    </div>
  );
};

export default Counter;
