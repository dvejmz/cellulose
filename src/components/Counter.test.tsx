import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import Counter from './Counter';

describe('<Counter />', () => {
  let counter: ShallowWrapper;

  beforeEach(() => {
    counter = shallow(
      <Counter
        id="total-paper"
        name="Total Paper"
        quantityUnit="sheets"
        quantity={9001}
      />);
  });

  it('renders ok', () => {
    expect(counter.exists()).toBeTruthy();
  });

  it('shows counter name', () => {
    expect(counter.find('.counter__name').text()).toBe('Total Paper')
  });

  it('shows counter quantity', () => {
    expect(counter.find('.counter__value').text()).toBe('9001');
  });

  it('shows counter quantity as a truncated decimal if decimal option is enabled', () => {
    counter = shallow(
      <Counter
        name="Paper"
        id="total-paper"
        quantityUnit="sheets"
        quantity={9001}
        showDecimals
      />);
    expect(counter.find('.counter__value').text()).toBe('9001.00');
  });

  it('shows counter unit', () => {
    expect(counter.find('.counter__unit').text()).toBe('sheets')
  });
});
