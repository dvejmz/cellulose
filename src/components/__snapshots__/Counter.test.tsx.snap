// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders 1`] = `
<div
  className="counter margin-top-right"
  data-test-id="counter-total-paper"
>
  <strong
    className="counter__name capitalise"
  >
    Total Paper
  </strong>
  :
   
  <span
    className="counter__value"
  >
    9001
  </span>
   
  <span
    className="counter__unit"
  >
    sheets
  </span>
</div>
`;
